crates/anvil-polkadot/tests/it/impersonation.rs

```diff
@@ -0,0 +1,199 @@
+use crate::utils::{TestNode, unwrap_response};
+use alloy_eips::BlockId;
+use alloy_primitives::{Address, U256};
+use alloy_rpc_types::TransactionRequest;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::{
+    api_server::revive_conversions::{AlloyU256, ReviveAddress},
+    config::{AnvilNodeConfig, SubstrateNodeConfig},
+};
+use polkadot_sdk::pallet_revive::evm::Account;
+use subxt::utils::H160;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_impersonate_account() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config.clone(), substrate_node_config).await.unwrap();
+
+    let latest_block = Some(BlockId::latest());
+
+    let alith_account = Account::from(subxt_signer::eth::dev::alith());
+    let alith_addr = Address::from(ReviveAddress::new(alith_account.address()));
+    let dest_addr = Address::random();
+    let dest_h160 = H160::from_slice(dest_addr.as_slice());
+    let transfer_amount = U256::from_str_radix("1600000000000000000", 10).unwrap();
+
+    // Create a random account with some balance.
+    unwrap_response::<()>(node.eth_rpc(EthRequest::SetAutomine(true)).await.unwrap()).unwrap();
+    let alith_initial_balance = node.get_balance(alith_account.address(), latest_block).await;
+    let dest_initial_balance = node.get_balance(dest_h160, latest_block).await;
+
+    let transaction =
+        TransactionRequest::default().value(transfer_amount).from(alith_addr).to(dest_addr);
+    let tx_hash = node.send_transaction(transaction, Some(1)).await.unwrap();
+    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+    let receipt_info = node.get_transaction_receipt(tx_hash).await;
+
+    // Assert on balances after first transfer.
+    let alith_final_balance = node.get_balance(alith_account.address(), latest_block).await;
```

Comment:
nit: Could use None for BlockId as it is expanded when converting to the Revive struct anyway.